docs/src/index.md

```diff
@@ -133,20 +133,30 @@ julia> autodiff(Forward, rosenbrock_inp, BatchDuplicated, BatchDuplicated(x, (dx
 
 If you pass in any temporary storage which may be involved in an active computation to a function you want to differentiate, you must also pass in a duplicated temporary storage for use in computing the derivatives. 
 
-```julia
+```jldoctest storage
 function f(x, tmp, n)
     tmp[1] = 1
     for i in 1:n
         tmp[1] *= x
     end
     tmp[1]
-end
+end;
+```
+
+```jldoctest storage
+Enzyme.autodiff(Reverse, f, Active(1.2), Const(Vector{Float64}(undef, 1)), Const(5))  # Incorrect [ returns (0.0,) ]
 
-# Incorrect [ returns (0.0,) ]
-Enzyme.autodiff(f, Active(1.2), Const(Vector{Float64}(undef, 1)), Const(5))
+# output
 
-# Correct [ returns (10.367999999999999,) == 1.2^4 * 5 ]
-Enzyme.autodiff(f, Active(1.2), Duplicated(Vector{Float64}(undef, 1), Vector{Float64}(undef, 1)), Const(5))
+((0.0, nothing, nothing),)
```

Comment:
So an optimization can turn this from using a buffer [as here] into just a simple register, e.g. here proving we are returning the value we just stored, we dont need to reload the memory, crcumventing the problem. Looks like recent Julia versions can do it for this case (which is cool). If you pass in the index as an arg for say just the load and not the store, it would stll err

Reply:
So how would you modify the example?

Reply:
Add an extra argument i with value Const(1), and use tmp[i] *= ... instead (but use tmp[1] elsewhere)

Reply:
done